packages/@aws-cdk/cloudformation-include/test/valid-templates.test.ts

```diff
@@ -657,6 +657,129 @@ describe('CDK Include', () => {
       cfnTemplate.getOutput('FakeOutput');
     }).toThrow(/Output with logical ID 'FakeOutput' was not found in the template/);
   });
+
+  test('parameters are replaced only when specified', () => {
+    new inc.CfnInclude(stack, 'includeTemplate', {
+      templateFile: testTemplateFilePath('bucket-with-parameters.json'),
+      parameterValues: {
+        BucketName: 'my-s3-bucket',
+      },
+    });
+
+    expect(stack).toMatchTemplate({
+      "Parameters": {
+        "CorsMaxAge": {
+          "Default": "3",
+          "Description": "the time in seconds that a browser will cache the preflight response",
+          "MaxValue": "300",
+          "MinValue": "0",
+          "AllowedValues": [1, 2, 3, 10, 100, 300],
+          "Type": "Number",
+          "NoEcho": "true",
+        },
+      },
+      "Resources": {
+        "Bucket": {
+          "Type": "AWS::S3::Bucket",
+          "Properties": {
+            "BucketName": "my-s3-bucket",
+            "CorsConfiguration": {
+              "CorsRules": [{
+                "AllowedMethods": [
+                  "GET",
+                  "POST",
+                ],
+                "AllowedOrigins": [
+                  "origin1",
+                  "origin2",
+                ],
+                "MaxAge": {
+                  "Ref": "CorsMaxAge",
+                },
+              }],
+            },
+          },
+        },
+      },
+    });
+  });
+
+  test('can replace parameters referenced in conditions and metadata', () => {
+    new inc.CfnInclude(stack, 'includeTemplate', {
+      templateFile: testTemplateFilePath('parameter-references.json'),
+      parameterValues: {
+        MyParam: 'my-s3-bucket',
+      },
+    });
+
+    expect(stack).toMatchTemplate({
+      "Transform": {
+        "Name": "AWS::Include",
+        "Parameters": {
+          "Location": "my-s3-bucket",
+        },
+      },
+      "Metadata": {
+        "Field": "my-s3-bucket",
+      },
+      "Conditions": {
+        "AlwaysFalse": {
+          "Fn::Equals": [ "my-s3-bucket", "Invalid?BucketName"],
+        },
+      },
+      "Resources": {
+        "Bucket": {
+          "Type": "AWS::S3::Bucket",
+          "Metadata": {
+            "Field": "my-s3-bucket",
+          },
+        },
+      },
+      "Outputs": {
+        "MyOutput": {
+          "Value": "my-s3-bucket",
+        },
+      },
+    });
+  });
+
+  test('can replace parameters in Fn::Sub', () => {
+    new inc.CfnInclude(stack, 'includeTemplate', {
+      templateFile: testTemplateFilePath('fn-sub-parameters.json'),
+      parameterValues: {
+        MyParam: 'my-s3-bucket',
+      },
+    });
+
+    expect(stack).toMatchTemplate({
+      "Resources": {
+        "Bucket": {
+          "Type": "AWS::S3::Bucket",
+          "Properties": {
+            "BucketName": {
+              "Fn::Sub": [
+                "${MyParameter}-my-s3-bucket",
```

Comment:
This is quite a weird test for such a simple thing, especially considering that it's a new file... can we use a file that has just `"Fn::Sub": "${Param}"`, and see what happens?